spring-boot-project/spring-boot-actuator-autoconfigure/src/main/java/org/springframework/boot/actuate/autoconfigure/metrics/MetricsAutoConfiguration.java

```diff
@@ -132,6 +139,19 @@ public SpringIntegrationMetrics springIntegrationMetrics(
 
 	}
 
+	@Configuration
+	@ConditionalOnClass({ RabbitTemplate.class, Channel.class })
+	@ConditionalOnBean(AbstractConnectionFactory.class)
+	@ConditionalOnProperty(prefix = "spring.rabbitmq", name = "metrics", matchIfMissing = true)
+	static class MetricsRabbitConfiguration {
+
+		@Bean
+		public RabbitMetrics rabbitMetrics(AbstractConnectionFactory connectionFactory) {
```

Comment:
that only works if there is one `AbstractConnectionFactory` in the context. What if there is more than one? Wouldn't  we want to monitor them all rather than failing?

Reply:
Makes sense. We could get all the `AbstractConnectionFactory` with their bean name as well (see below).